aiida_optimize/_optimization_workchain.py

```diff
@@ -151,7 +155,10 @@ def finalize(self):  # pylint: disable=inconsistent-return-statements
         with self.optimizer() as opt:
             if not opt.is_finished_ok:
                 return self.exit_codes.ERROR_ENGINE_FAILED
-            optimal_process_output = opt.result_value
+            optimal_process_input = opt.result_input_value
+            assert optimal_process_input.is_stored
+            self.out('optimal_process_input', optimal_process_input)
```

Comment:
If we make it `required=False`, I guess this should only be executed `if optimal_process_input is not None`. That'll allow the engine to just return `None` on the `result_input_value`.

Reply:
But it would also be ok to keep the output required - at least until we have an engine that can't provide it.

Reply:
We'll have to handle this in `conftest.py` as well...

Reply:
Right - or we just make it `required=True` for now.